packages/ListBox/src/components/Trigger/Trigger.js

```diff
@@ -154,62 +122,98 @@ export default function Trigger(props) {
       type: useListBox.types.clear,
       payload: {
         isOpen: false,
-        onChangeFn: invokeOnChange(state.onChange, "list-box:clear"),
+        onChangeFn: invokeOnChange(onChange, "list-box:clear"),
       },
     });
   };
 
+  const hasRenderTrigger = typeof children === "function";
+
+  const shouldHideClearButton =
+    !hasClearButton || hasRenderTrigger || isReadOnly || (hasFooter && isOpen) || selectedOptions.length === 0;
+
+  const shouldHideCaret = hasRenderTrigger || isInline;
+
   function renderLabel() {
-    return state.isInline ? (
+    return isInline || isReadOnly ? (
       <Label
-        activeOption={state.options[state.activeOption]}
+        activeOption={options[activeOption]}
         hasImplicitAll={hasImplicitAll}
+        id={`${idListBox}__label`}
+        isDisabled={isDisabled}
         isMulti={isMulti}
-        options={state.options}
-        placeholder={placeholder}
-        selectedOptions={state.selectedOptions}
         label={label}
+        options={options}
+        placeholder={placeholder}
+        selectedOptions={selectedOptions}
       />
     ) : (
       <RawButton
-        id={triggerButtonId.current}
+        {...getDOMAttributesForListBoxButton({
+          idListBox,
+          idFormLabel: idFormLabel.current,
+        })}
+        {...morePropsForTrigger}
+        data-pka-anchor="list-box-trigger"
+        isDisabled={isDisabled}
         onClick={handleClick}
-        ref={refTrigger}
         onKeyUp={handleKeyUp}
-        isDisabled={isDisabled}
-        data-pka-anchor="list-box-trigger"
-        aria-haspopup
-        aria-describedby={formElementLabelDescribedBy}
-        aria-labelledby={triggerButtonId.current}
+        ref={refTrigger}
       >
-        {refLabel && refLabel.current ? (
-          <sc.VisuallyHiddenFormLabel>{refLabel.current.innerText}</sc.VisuallyHiddenFormLabel>
-        ) : null}
```

Comment:
This is now taken care of by `aria-labelledby` in `getDOMAttributesForListBoxButton()`